### PR TITLE
Add Acquire service (first pass)

### PR DESCRIPTION
**Pull request by jdenen**

### CODE REVIEW

File: apps/service_acquire/lib/service_acquire/presto.ex [hunk 1]
```diff
@@ -0,0 +1,27 @@
+defmodule Acquire.Presto do
```
Comment (bbalser) [inline apps/service_acquire/lib/service_acquire/presto.ex]: Should we use a more generic term for the name of the behaviour, Acquire.Query or something like that so behaviour is not tied to presto

Comment (jeffgrunewald) [reply apps/service_acquire/lib/service_acquire/presto.ex]: `Acquire.DataStore`

Comment (jeffgrunewald) [reply apps/service_acquire/lib/service_acquire/presto.ex]: Or if you prefer later with more options: `Acquire.DB.Relational` vs `Acquire.DB.Graph`

Comment (bbalser) [reply apps/service_acquire/lib/service_acquire/presto.ex]: I am good with any of those.

File: apps/service_acquire/lib/service_acquire/query/where/function.ex [hunk 2]
```diff
@@ -0,0 +1,66 @@
+defmodule Acquire.Query.Where.Function do
+  use Definition, schema: Acquire.Query.Where.Function.V1
+
+  @type t :: %__MODULE__{
+          function: String.t(),
+          args: [term]
+        }
+
+  defstruct [:function, :args]
```
Comment (jeffgrunewald) [inline apps/service_acquire/lib/service_acquire/query/where/function.ex]: Version?

File: apps/service_acquire/lib/service_acquire/query/where/or.ex [hunk 3]
```diff
@@ -0,0 +1,42 @@
+defmodule Acquire.Query.Where.Or do
+  use Definition, schema: Acquire.Query.Where.Or.V1
+  defstruct [:conditions]
```
Comment (jeffgrunewald) [inline apps/service_acquire/lib/service_acquire/query/where/or.ex]: Version?

File: apps/service_acquire/lib/service_acquire/query/where/parameter.ex [hunk 4]
```diff
@@ -0,0 +1,22 @@
+defmodule Acquire.Query.Where.Parameter do
+  use Definition, schema: Acquire.Query.Where.Parameter.V1
+
+  @type t :: %__MODULE__{value: term}
+
+  defstruct [:value]
```
Comment (jeffgrunewald) [inline apps/service_acquire/lib/service_acquire/query/where/parameter.ex]: Version?

File: apps/service_acquire/lib/service_acquire/query.ex [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            where: nil
+
+  defimpl Acquire.Queryable, for: __MODULE__ do
+    def parse_statement(query) do
+      fields = Enum.join(query.fields, ", ")
+      limit = limit_statement(query.limit)
+      where = where_statement(query.where)
+
+      ["SELECT", fields, "FROM", query.table, where, limit]
+      |> Enum.filter(& &1)
```
[the remark this answers is not in the document]
Comment (jeffgrunewald) [reply apps/service_acquire/lib/service_acquire/query.ex]: Validated this was the case.

File: apps/service_acquire/lib/service_acquire/query.ex [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    defp limit_statement(n), do: "LIMIT #{n}"
+  end
+
+  def from_params(%{"dataset" => dataset} = params) do
+    subset = Map.get(params, "subset", "default")
+
+    fields =
+      Map.get(params, "fields", "*")
+      |> String.split(",", trim: true)
+      |> Enum.map(&String.trim/1)
```
[the remark this answers is not in the document]
Comment (jeffgrunewald) [reply apps/service_acquire/lib/service_acquire/query.ex]: validated the functionality of `trim: true`; it seems like a miss that this doesn't trim whitespace. oh well.

File: apps/service_acquire/lib/service_acquire/query/where.ex [hunk 7]
```diff
@@ -0,0 +1,40 @@
+defmodule Acquire.Query.Where do
+  alias Acquire.Query.Where.{And, FilterParser}
+  alias Acquire.Query.Where.Bbox
```
[the remark this answers is not in the document]
Comment (jdenen) [reply apps/service_acquire/lib/service_acquire/query/where.ex]: Yeah this was a miss caused by refactoring. Nice catch.

File: apps/service_acquire/lib/service_acquire/query/where/and.ex [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  end
+end
+
+defmodule Acquire.Query.Where.And.V1 do
+  use Definition.Schema
+
+  @impl true
+  def s do
+    schema(%Acquire.Query.Where.And{
+      conditions: conditions()
```
[the remark this answers is not in the document]
Comment (jdenen) [reply apps/service_acquire/lib/service_acquire/query/where/and.ex]: Yes

File: apps/definition/lib/definition/schema/type.ex [hunk 9]
```diff
@@ -34,7 +34,7 @@ defmodule Definition.Schema.Type do
   end
 
   @spec access_path() :: spec
-  def access_path() do
+  def access_path do
```
[the remark this answers is not in the document]
Comment (jdenen) [reply apps/definition/lib/definition/schema/type.ex]: The formatter doesn't add the empty parens to zero arity functions. 🤷‍♂

File: apps/definition/lib/definition/schema/validation.ex [hunk 10]
```diff
@@ -136,4 +136,24 @@ defmodule Definition.Schema.Validation do
...
+  end
+
+  def table_name?(_), do: false
+
+  @doc """
+  Evaluates whether input is a positive integer.
+  """
+  @spec pos_integer?(term) :: boolean
+  def pos_integer?(input) when is_integer(input) and input > 0, do: true
+  def pos_integer?(_), do: false
```
[the remark this answers is not in the document]
Comment (jdenen) [reply apps/definition/lib/definition/schema/validation.ex]: Yes, we do that, but it should be done before creating a struct that uses this validation. A struct that expects a positive integer should be given a positive integer only.

File: apps/service_acquire/lib/service_acquire/query.ex (hunk 6 above)
[the remark this answers is not in the document]
Comment (jdenen) [reply apps/service_acquire/lib/service_acquire/query.ex]: I agree, it's confusing.

File: apps/definition/lib/definition/schema/validation.ex [hunk 11]
```diff
@@ -136,4 +136,24 @@ defmodule Definition.Schema.Validation do
...
   def not_nil?(nil), do: false
   def not_nil?(_), do: true
+
+  @doc """
+  Evaluates whether input matches an acceptable table name pattern.
+  """
+  @spec table_name?(term) :: boolean
+  def table_name?(input) when is_binary(input) do
+    case String.split(input, "__", trim: true) do
+      [_, _] -> true
```
[the remark this answers is not in the document]
Comment (jdenen) [reply apps/definition/lib/definition/schema/validation.ex]: This just validates as part of `Norm`. It does no coercing, and that should be handled before a struct is initialized.

**Comment by jdenen** [comment]
Nice catch on the directory name issue. I changed the schema module names to imply there is no versioning. I'm using `Definition` to take advantage of the `new/1`/`new!/1` functions and `Norm` validation. I do not envision us using versioning or migration as part of these structs since they aren't stored anywhere.

**Comment by jdenen** [comment]
Things left to do with Acquire:

- [ ] Brook setup correctly
- [ ] Transformation includes field for `subset` name
- [ ] Temporal filters
- [ ] Passing e2e tests

**Comment by jdenen** [comment]
Talked to @bbalser; merging what we've got so far and will pick up the rest in future PRs.

**Pull request merged**